### DISCUSSION TITLE
Data roll up over 1m at Floor level

### DISCUSSION THREAD

**Discussion by ShivaniSK** [finos/TimeBase-CE#229] [Q&A]
Hi,
We are currently querying timebase to roll up data to minute level using below query
```
select sum{} (value) from test_data over TIME (1m) where timestamp BETWEEN 1661472000000 
AND 1661558399000 AND SYMBOL IN ('ABC')
```
And we are getting the output where the data is rolled up at minute level and the minute is at ceiling level.
E.g
**Input data**
```
Timestamp    value
01:01:20          2
01:01:54          3
02:02: 23         4
```

**Output after 1 min roll up**
```
Timestamp    value
01:02               5
02:02               4
```

Which means **`01:02         5`** indicates the data between 01:01 to 01:02.
Is it possible to roll up minute data at "Floor level" ? Which means the records between 01:01 to 01:02 should be rolled up to **01:01** instead of **01:02**

**Comment by alex-karpovich** [comment]
Greetings @ShivaniSK, 
```over time(...)``` aggregation uses end of interval as timestamp for the message. 
Overall, if periodical data stored in TimeBase, all timestamps considered as "end" time. 
Typical sample from financial domain is bar messages, when stored in TimeBase stream, they timestamps considered as "close time".
